src/System.Net.Http/src/uap/System/Net/HttpClientHandler.cs

```diff
@@ -684,6 +714,63 @@ private static bool InitRTNoCacheSupported()
                 "NoCache");
         }
 
+        private void RTServerCertificateCallback(RTHttpBaseProtocolFilter sender, RTHttpServerCustomValidationRequestedEventArgs args)
+        {
+            // Convert WinRT certificate to .NET certificate.
+            X509Certificate2 serverCert = ConvertCertificate(args.ServerCertificate);
+
+            // Create .NET X509Chain from the WinRT information. We need to rebuild
+            // the chain since WinRT only gives us an array of intermediate certificates.
+            var serverChain = new X509Chain();
+            foreach (RTCertificate cert in args.ServerIntermediateCertificates)
+            {
+                serverChain.ChainPolicy.ExtraStore.Add(ConvertCertificate(cert));
+            }
+            serverChain.Build(serverCert);
+
+            // Determine chain and name errors from WinRT information.
+            SslPolicyErrors sslPolicyErrors = SslPolicyErrors.None;
+            if (serverChain.ChainStatus.Length > 0)
+            {
+                sslPolicyErrors |= SslPolicyErrors.RemoteCertificateChainErrors;
+            }
+            foreach (RTChainValidationResult result in args.ServerCertificateErrors)
```

Comment:
>The chain building part here should match SslStream.

@bartonjs What does this mean exactly?  Does it mean that using Chain.Build() as I am doing here sufficient?  Or we need to call the low-level SSPI stuff as is doing in SslStream and WinHttpHandler. And by the way, that code also calls Chain.Build().  The only time it uses SSPI stuff is to figure out the "name mismatch" determination.  And we already know that because the WinRT APIs return that information already (a feature missing in .NET X509Chain.Build because it doesn't take a DNS server name parameter).

So, what should I do regarding my PR and how I am "converting" from WinRT certificate information to .NET X509Chain?